test/uk/gov/hmrc/plasticpackagingtax/registration/connectors/EmailVerificationConnectorSpec.scala

```diff
@@ -0,0 +1,271 @@
+                                             credId,
+                                             toJson(validResponse).toString
+      )
+
+      val res = await(connector.getStatus(credId))
+
+      res.left.value.getMessage must include("Failed to retrieve email verification status")
+    }
+
+    "service returns not found status code" in {
+      giveGetEmailVerificationDetailsReturns(Status.NOT_FOUND, credId)
+
+      val res = await(connector.getStatus(credId))
+
+      res.value mustBe None
+    }
+
+    "service returns invalid response" in {
+      giveGetEmailVerificationDetailsReturns(Status.OK, credId, "someRubbish")
+
+      val res = await(connector.getStatus(credId))
+
+      res.left.value.getMessage must include("Failed to retrieve email verification status")
+    }
+  }
+
+  "submitPasscode" when {
+    val journeyId             = "credId"
+    val verifyPasscodeRequest = VerifyPasscodeRequest("passcode", "email")
+
+    "returns complete" in {
+      val validResponse = "{\"status\":\"complete\"}"
+      givenPostToSubmitPascode(OK, journeyId, validResponse)
+
+      val res = await(connector.verifyPasscode(journeyId, verifyPasscodeRequest))
+
+      res.value mustBe "complete"
+      getTimer("ppt.email.verification.verify.passcode.timer").getCount mustBe 1
+    }
+
+    "returns bad request" in {
+      val validResponse = "{\"status\":\"incorrectPasscode\"}"
+      givenPostToSubmitPascode(BAD_REQUEST, journeyId, validResponse)
+
+      val res = await(connector.verifyPasscode(journeyId, verifyPasscodeRequest))
+
+      res.value mustBe "incorrectPasscode"
+    }
+
+    "returns forbidden" in {
+      val validResponse = "{\"status\":\"tooManyAttempts\"}"
+      givenPostToSubmitPascode(FORBIDDEN, journeyId, validResponse)
+
+      val res = await(connector.verifyPasscode(journeyId, verifyPasscodeRequest))
+
+      res.value mustBe "tooManyAttempts"
+    }
+
+    "returns not found" in {
+      val validResponse = "{\"status\":\"passcodeNotFound\"}"
+      givenPostToSubmitPascode(FORBIDDEN, journeyId, validResponse)
+
+      val res = await(connector.verifyPasscode(journeyId, verifyPasscodeRequest))
+
+      res.value mustBe "passcodeNotFound"
+    }
+
+    "returns internal server error" in {
+      givenPostToSubmitPascode(INTERNAL_SERVER_ERROR, journeyId)
+
+      val res = await(connector.verifyPasscode(journeyId, verifyPasscodeRequest))
+
+      res.left.value.getMessage must include("Failed to verify passcode, status: 500, error: ")
+    }
+  }
+
+  private def giveGetEmailVerificationDetailsReturns(
+    status: Int,
+    credId: String,
+    body: String = ""
+  ) =
+    stubFor(
+      get(urlMatching(s"/email-verification/verification-status/$credId"))
+        .willReturn(
+          aResponse()
+            .withStatus(status)
+            .withBody(body)
+        )
+    )
+
+  private def givenPostToEmailVerificationReturns(status: Int, body: String = "") =
```

Comment:
Please remove the default value for body as it is never used.

Reply:
Removed the default value